In [1]:
#Import libraries
import pandas as pd
import numpy as np
import os

In [2]:
path= '/Users/zacharygreen/Desktop/instacart analysis/data'

In [3]:
df_ords=pd.read_csv(os.path.join(path, 'prepared data', 'orders_wrangled.csv'), index_col=False)

In [4]:
df_ords.describe()

Unnamed: 0      order_id       user_id  order_number  \
count  3.421083e+06  3.421083e+06  3.421083e+06  3.421083e+06   
mean   1.710541e+06  1.710542e+06  1.029782e+05  1.715486e+01   
std    9.875817e+05  9.875817e+05  5.953372e+04  1.773316e+01   
min    0.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00   
25%    8.552705e+05  8.552715e+05  5.139400e+04  5.000000e+00   
50%    1.710541e+06  1.710542e+06  1.026890e+05  1.100000e+01   
75%    2.565812e+06  2.565812e+06  1.543850e+05  2.300000e+01   
max    3.421082e+06  3.421083e+06  2.062090e+05  1.000000e+02   

       orders_day_of_week  order_hour_of_day  days_since_prior_order  
count        3.421083e+06       3.421083e+06            3.214874e+06  
mean         2.776219e+00       1.345202e+01            1.111484e+01  
std          2.046829e+00       4.226088e+00            9.206737e+00  
min          0.000000e+00       0.000000e+00            0.000000e+00  
25%          1.000000e+00       1.000000e+01            4.000000e+00  
50%          3.000000e+00       1.300000e+01            7.000000e+00  
75%          5.000000e+00       1.600000e+01            1.500000e+01  
max          6.000000e+00       2.300000e+01            3.000000e+01

In [5]:
#check for mixed types
for col in df_ords.columns.tolist():
  weird = (df_ords[[col]].applymap(type) != df_ords[[col]].iloc[0].apply(type)).any(axis = 1)
  if len (df_ords[weird]) > 0:
    print (col)

No mixed data types

In [6]:
df_ords.isnull().sum()

Unnamed: 0                     0
order_id                       0
user_id                        0
order_number                   0
orders_day_of_week             0
order_hour_of_day              0
days_since_prior_order    206209
dtype: int64

206,209 values, most likely because some customers were ordering for the first time. We're going to delete these, because we don't need them. 

In [7]:
df_ords.dropna(inplace=True)

In [8]:
df_ords.shape

(3214874, 7)

In [9]:
df_dups=df_ords[df_ords.duplicated()]

In [10]:
df_dups

Empty DataFrame
Columns: [Unnamed: 0, order_id, user_id, order_number, orders_day_of_week, order_hour_of_day, days_since_prior_order]
Index: []

No duplicate values 

In [11]:
loc.days_since_prior_order

NameError: name 'loc' is not defined

In [12]:
df_ords.head(1000)

Unnamed: 0  order_id  user_id  order_number  orders_day_of_week  \
1              1   2398795        1             2                   3   
2              2    473747        1             3                   3   
3              3   2254736        1             4                   4   
4              4    431534        1             5                   4   
5              5   3367565        1             6                   2   
...          ...       ...      ...           ...                 ...   
1065        1065    939800       70             6                   4   
1066        1066   2509231       70             7                   4   
1067        1067   2373040       70             8                   6   
1068        1068   2950172       70             9                   4   
1069        1069   2848644       70            10                   4   

      order_hour_of_day  days_since_prior_order  
1                     7                    15.0  
2                    12                    21.0  
3                     7                    29.0  
4                    15                    28.0  
5                     7                    19.0  
...                 ...                     ...  
1065                 16                    11.0  
1066                 15                    30.0  
1067                 12                     9.0  
1068                 19                    12.0  
1069                 20                    14.0  

[1000 rows x 7 columns]

In [13]:
print(df_ords['days_since_prior_order'].head(1000))

1       15.0
2       21.0
3       29.0
4       28.0
5       19.0
        ... 
1065    11.0
1066    30.0
1067     9.0
1068    12.0
1069    14.0
Name: days_since_prior_order, Length: 1000, dtype: float64


In [15]:
# Assuming you have a DataFrame named df_ords with a column named 'days_since_prior_order'

# Convert the column to a string representation and print it
print(df_ords['days_since_prior_order'].to_string(index=False))


IOPub data rate exceeded.
The notebook server will temporarily stop sending output
to the client in order to avoid crashing it.
To change this limit, set the config variable
`--NotebookApp.iopub_data_rate_limit`.

Current values:
NotebookApp.iopub_data_rate_limit=1000000.0 (bytes/sec)
NotebookApp.rate_limit_window=3.0 (secs)

